Assets/Scripts/Audio/AudioDistanceController.cs

```diff
@@ -0,0 +1,62 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+
+/// <summary>
+/// This script is for making audio player louder/quieter depending on the distance to the audio
+/// source. 
+/// 
+/// Make sure to look at the object's parent and assaign the correct localScale.
+/// </summary>
+
+
+public class AudioDistanceController : MonoBehaviour {
+
+    public float maxVolume;
+
+    //The radius of the circle collider is determined by the local scale (For example if 
+    // the parent container was scaled to 0.5 and the radius was 15, the actuall global radius 
+    // would only be 7.5. 
+    public float localScale;
+    private CircleCollider2D audioRangeCollider;
+    private AudioSource audioSource;
+    private List<GameObject> dolls = new List<GameObject>();
+
+    // Use this for initialization
+    void Start() {
+        audioRangeCollider = GetComponent<CircleCollider2D>();
+        audioSource = GetComponent<AudioSource>();
+    }
+
+    void Update() {
+        //Check if the list of dolls includes the player doll, else the volume of the object should be 0
+        GameObject player = dolls.Find(x => x.CompareTag("Player"));
+        if (player != null) {
+            changeVolumeByDistance(player);
+        } else {
+            audioSource.volume = 0;
+        }
+    }
+
+    //Add the possible player to the list of dolls
+    void OnTriggerEnter2D(Collider2D other) {
```

Comment:
Again same as above, it doesn't work like that because of the nature of the player doll and non-player doll and switching out.
Also, I tried using the onTriggerStay and then checking if the doll inside the collider is a player doll or not, but this doesn't work because of that problem and also the problem of competing preference. If we add a check to if the doll is a player or not in the onTriggerStay, then if there is a doll that is a player and a doll that isn't inside the collider then one will try to change the audio to 0, while the other will try to play the audio.